File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        cur = cnt = 0
        mp = {}
        mp[0] = 1
        for n in nums:
            cur += n
            diff = cur - k
            if diff in mp:
                cnt += mp[diff]
            mp[cur] = mp.get(cur, 0) + 1
        return cnt

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    const freq = { 0: 1 }
    let sum = 0, count = 0
    for (const n of nums) {
        sum += n
        const diff = sum - k
        if (diff in freq) {
            count += freq[diff]
        }
        freq[sum] = freq[sum] + 1 || 1
    }
    return count
};